src/mlpack/tests/sdp_primal_dual_test.cpp

```diff
@@ -557,92 +558,141 @@ BOOST_AUTO_TEST_CASE(CorrelationCoeffToySdp)
-//  arma::mat distances;
-//  KNN knn(origData);
-//  knn.Search(numNeighbors, neighbors, distances);
-
-//  SDP<arma::sp_mat> sdp(numPoints, numNeighbors * numPoints, 1);
-//  sdp.C().eye(numPoints, numPoints);
-//  sdp.C() *= -1;
-//  sdp.DenseA()[0].ones(numPoints, numPoints);
-//  sdp.DenseB()[0] = 0;
-
-//  for (size_t i = 0; i < neighbors.n_cols; ++i)
-//  {
-//    for (size_t j = 0; j < numNeighbors; ++j)
-//    {
-//      // This is the index of the constraint.
-//      const size_t index = (i * numNeighbors) + j;
-
-//      arma::sp_mat& aRef = sdp.SparseA()[index];
-//      aRef.zeros(numPoints, numPoints);
-
-//      // A_ij(i, i) = 1.
-//      aRef(i, i) = 1;
-
-//      // A_ij(i, j) = -1.
-//      aRef(i, neighbors(j, i)) = -1;
-
-//      // A_ij(j, i) = -1.
-//      aRef(neighbors(j, i), i) = -1;
-
-//      // A_ij(j, j) = 1.
-//      aRef(neighbors(j, i), neighbors(j, i)) = 1;
-
-//      // The constraint b_ij is the distance between these two points.
-//      sdp.SparseB()[index] = distances(j, i);
-//    }
-//  }
-
-//  return sdp;
-// }
-
-// /**
-// * Maximum variance unfolding
-// *
-// * Test doesn't work, because the constraint matrices are not linearly
-// * independent.
-// */
-// BOOST_AUTO_TEST_CASE(SmallMvuSdp)
-// {
-//  const size_t n = 20;
-
-//  arma::mat origData(3, n);
-
-//  // sample n random points on 3-dim unit sphere
-//  GaussianDistribution gauss(3);
-//  for (size_t i = 0; i < n; i++)
-//  {
-//    // how european of them
-//    origData.col(i) = arma::normalise(gauss.Random());
-//  }
-
-//  auto sdp = ConstructMvuSDP(origData, 5);
-
-//  PrimalDualSolver<SDP<arma::sp_mat>> solver(sdp);
-//  arma::mat X, Z;
-//  arma::vec ysparse, ydense;
-//  const auto p = solver.Optimize(X, ysparse, ydense, Z);
-//  BOOST_REQUIRE(p.first);
-// }
+/**
+ * Maximum variance unfolding (MVU) SDP to learn the unrolled gram matrix. For
+ * the SDP formulation, see:
+ *
+ *   Unsupervised learning of image manifolds by semidefinite programming.
+ *   Kilian Weinberger and Lawrence Saul. CVPR 04.
+ *   http://repository.upenn.edu/cgi/viewcontent.cgi?article=1000&context=cis_papers
+ *
+ * Here is another paper which clarifies more specific SDP formulation of MVU problem.
+ *
+ *   Scalable semidefinite manifold learning
+ *   Nikolaos Vasiloglou, Alexander G. Gray, David V. Anderson.
+ *   2008 IEEE Workshop on Machine Learning for Signal Processing
+ *
+ * @param origData origDim x numPoints
+ * @param numNeighbors
+ */
+static inline SDP<arma::sp_mat> ConstructMvuSDP(const arma::mat& origData,
+                                              size_t numNeighbors)
+{
+  const size_t numPoints = origData.n_cols;
+
+  assert(numNeighbors <= numPoints);
+
+  arma::Mat<size_t> neighbors;
+  arma::mat distances;
+  KNN knn(origData);
+  knn.Search(numNeighbors, neighbors, distances);
+
+  // how to construct the matrices A, b, and C? Plz refer to the above paper.
+  std::set<std::pair<int, int>> neighbors_set;
```

Comment:
A minor style issue... we should call this `neighborsSet` not `neighbors_set` (i.e., no underscores in the variable names, we use camelcase naming instead).